Remove client side enforced min execution time

**What is the expected feature or enhancement?**

The server side API for POST /jobs no longer enforces a min value of 300 for `max_execution_time` but the client still does:

https://github.com/Qiskit/qiskit-ibm-runtime/blob/7752b68c749cc434feeacb5caf72d02de91aed46/qiskit_ibm_runtime/options/options.py#L173

The client should leave the validation to the server side and just handle errors that arise from that validation rather than impose it's own limits, unless there is some good reason, but here it just seems redundant and in this case no longer applicable.


**Acceptance criteria**

The min execution time validation is removed.

cc @riley2012 @jyu00

I'm using qiskit-ibm-runtime 0.12.0 and when I try to submit a Sampler job with `max_execution_time=1` it fails here:

```
        if not execution_time is None:
            if (
                execution_time < Options._MIN_EXECUTION_TIME
                or execution_time > Options._MAX_EXECUTION_TIME
            ):
>               raise ValueError(
                    f"max_execution_time must be between "
                    f"{Options._MIN_EXECUTION_TIME} and {Options._MAX_EXECUTION_TIME} seconds."
                )
E               ValueError: max_execution_time must be between 300 and 28800 seconds.

.tox/staging/lib/python3.8/site-packages/qiskit_ibm_runtime/options/options.py:176: ValueError
```

I'm trying to add some integration tests which make sure that the system times out and cancels jobs as expected when their max_execution_time is reached.

--- a/qiskit_ibm_runtime/options/options.py
+++ b/qiskit_ibm_runtime/options/options.py
@@ -90,7 +90,6 @@ class Options:
     _MAX_OPTIMIZATION_LEVEL = 3
     _MAX_RESILIENCE_LEVEL_ESTIMATOR = 3
     _MAX_RESILIENCE_LEVEL_SAMPLER = 1
-    _MIN_EXECUTION_TIME = 300
     _MAX_EXECUTION_TIME = 8 * 60 * 60  # 8 hours for real device
 
     optimization_level: Optional[int] = None
@@ -168,14 +167,10 @@ class Options:
         ResilienceOptions.validate_resilience_options(options.get("resilience"))
         TranspilationOptions.validate_transpilation_options(options.get("transpilation"))
         execution_time = options.get("max_execution_time")
-        if not execution_time is None:
-            if (
-                execution_time < Options._MIN_EXECUTION_TIME
-                or execution_time > Options._MAX_EXECUTION_TIME
-            ):
+        if execution_time is not None:
+            if execution_time > Options._MAX_EXECUTION_TIME:
                 raise ValueError(
-                    f"max_execution_time must be between "
-                    f"{Options._MIN_EXECUTION_TIME} and {Options._MAX_EXECUTION_TIME} seconds."
+                    f"max_execution_time must be below " f"{Options._MAX_EXECUTION_TIME} seconds."
                 )
 
         EnvironmentOptions.validate_environment_options(options.get("environment"))

--- a/qiskit_ibm_runtime/utils/qctrl.py
+++ b/qiskit_ibm_runtime/utils/qctrl.py
@@ -32,14 +32,10 @@ def validate(options: Dict[str, Any]) -> None:
     # Default validation otherwise.
     TranspilationOptions.validate_transpilation_options(options.get("transpilation"))
     execution_time = options.get("max_execution_time")
-    if not execution_time is None:
-        if (
-            execution_time < Options._MIN_EXECUTION_TIME
-            or execution_time > Options._MAX_EXECUTION_TIME
-        ):
+    if execution_time is not None:
+        if execution_time > Options._MAX_EXECUTION_TIME:
             raise ValueError(
-                f"max_execution_time must be between "
-                f"{Options._MIN_EXECUTION_TIME} and {Options._MAX_EXECUTION_TIME} seconds."
+                f"max_execution_time must be below " f"{Options._MAX_EXECUTION_TIME} seconds."
             )
 
     EnvironmentOptions.validate_environment_options(options.get("environment"))

--- a/test/integration/test_job.py
+++ b/test/integration/test_job.py
@@ -131,12 +131,6 @@ class TestIntegrationJob(IBMIntegrationJobTestCase):
         self.assertEqual(job._api_client.job_get(job.job_id())["cost"], job_max_execution_time)
 
     @run_integration_test
-    def test_invalid_max_execution_time_fails(self, service):
-        """Test that program fails when max_execution_time is less than 300."""
-        with self.assertRaises(ValueError):
-            self._run_program(service, max_execution_time=299)
-
-    @run_integration_test
     @production_only
     def test_cancel_job_queued(self, service):
         """Test canceling a queued job."""

--- a/test/unit/test_ibm_primitives.py
+++ b/test/unit/test_ibm_primitives.py
@@ -688,7 +688,6 @@ class TestPrimitives(IBMTestCase):
     def test_max_execution_time_options(self):
         """Test transpilation options."""
         options_dicts = [
-            {"max_execution_time": Options._MIN_EXECUTION_TIME - 1},
             {"max_execution_time": Options._MAX_EXECUTION_TIME + 1},
         ]
         session = MagicMock(spec=MockSession)

--- a/test/unit/test_ibm_primitives.py
+++ b/test/unit/test_ibm_primitives.py
@@ -707,7 +707,7 @@ class TestPrimitives(IBMTestCase):
                     inst = cls(session=session, options=opts_dict)
                     inst.run(self.qx, observables=self.obs)
                 self.assertIn(
-                    "max_execution_time must be between 300 and 28800 seconds",
+                    "max_execution_time must be below 28800 seconds",
                     str(exc.exception),
                 )
 

